tron/actioncommand.py

```diff
@@ -41,48 +27,58 @@ class ActionCommand(object):
       done      (when the command is finished)
     """
 
-    COMPLETE = ActionState('complete')
-    FAILSTART = ActionState('failstart')
-    EXITING = ActionState('exiting', close=COMPLETE)
-    RUNNING = ActionState('running', exit=EXITING)
-    PENDING = ActionState('pending', start=RUNNING, exit=FAILSTART)
+    PENDING = 'pending'
+    RUNNING = 'running'
+    EXITING = 'exiting'
+    COMPLETE = 'complete'
+    FAILSTART = 'failstart'
+
+    STATE_MACHINE = Machine(
+        PENDING,
+        **{
+            PENDING: {'start': RUNNING, 'exit': FAILSTART},
+            RUNNING: {'exit': EXITING},
+            EXITING: {'close': COMPLETE},
+        }
+    )
 
     STDOUT = '.stdout'
     STDERR = '.stderr'
 
     def __init__(self, id, command, serializer=None):
+        super().__init__()
         self.id = id
         self.command = command
-        self.machine = state.StateMachine(self.PENDING, delegate=self)
+        self.machine = Machine.from_machine(ActionCommand.STATE_MACHINE)
         self.exit_status = None
         self.start_time = None
         self.end_time = None
-        self.stdout = filehandler.NullFileHandle
-        self.stderr = filehandler.NullFileHandle
         if serializer:
             self.stdout = serializer.open(self.STDOUT)
             self.stderr = serializer.open(self.STDERR)
+        else:
+            self.stdout = filehandler.NullFileHandle
+            self.stderr = filehandler.NullFileHandle
 
     @property
     def state(self):
         return self.machine.state
 
-    @property
-    def attach(self):
-        return self.machine.attach
+    def transition_and_notify(self, target):
```

Comment:
These functions used to be able to return false, but now I think they will return an implicit None, is that OK?

Reply:
i'm ok with that. even if there are too many things falsey in python, `None` is one i agree with.